elastic/integrations
# qualys_vmdr.asset_host_detection: Capture issue with XML parsing into error.message

## Proposed commit message
Captures 2 scenarios into `error.message` to give better traceability.
- Invalid XML
- When XML response doesn't match with XSD defined in the input.



## Checklist


## How to test this PR locally


1. The system tests should succeed with current `default` configuration.
      Run:
      `eval "$(elastic-package stack shellinit)" && elastic-package test system --generate  -v --data-streams=asset_host_detection`
      Returns: 
      ```
      --- Test results for package: qualys_vmdr - START ---
      ╭─────────────┬──────────────────────┬───────────┬───────────┬────────┬───────────────╮
      │ PACKAGE     │ DATA STREAM          │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
      ├─────────────┼──────────────────────┼───────────┼───────────┼────────┼───────────────┤
      │ qualys_vmdr │ asset_host_detection │ system    │ default   │ PASS   │ 41.089226334s │
      ╰─────────────┴──────────────────────┴───────────┴───────────┴────────┴───────────────╯
      --- Test results for package: qualys_vmdr - END   ---
      Done
      ```

2. The system tests should fail (false positive - due to presence of `error.message`) with new configurations. Copy files from [qualys-system-test.zip](https://github.com/user-attachments/files/18022167/qualys-system-test.zip) into respective places:
    1. Contents of `qualys_vmdr_failure-config.yml` at the top of existing file `packages/qualys_vmdr/_dev/deploy/docker/files/config.yml`
    2. Files `test-fail-invalid-xml-config.yml` and `test-fail-mismatch-xsd-config.yml` into `packages/qualys_vmdr/data_stream/asset_host_detection/_dev/test/system`. 
    4. Run: `eval "$(elastic-package stack shellinit)" && elastic-package test system --generate  -v --data-streams=asset_host_detection`
    Returns:
      ```
      2024/12/05 16:23:49 DEBUG DELETE https://127.0.0.1:5601/api/fleet/epm/packages/qualys_vmdr/5.5.0
      --- Test results for package: qualys_vmdr - START ---
      FAILURE DETAILS:
      qualys_vmdr/asset_host_detection fail-invalid-xml:
      [0] found error.message in event: [decode_xml error:failed to unmarshal XML document: XML syntax error on line 7: unexpected EOF:<?xml version="1.0" encoding="UTF-8"?>
      <!DOCTYPE HOST_LIST_VM_DETECTION_OUTPUT SYSTEM "https://qualysapi.qualys.com/api/2.0/fo/asset/host/vm/detection/dtd/output.dtd">
      <HOST_LIST_VM_DETECTION_OUTPUT>
        <RESPONSE>
          <DATETIME>2023-07-04T06:51:41Z</DATETIME>
          <HOST_LIST>
            <HOST> Processor conditional with tag cel_failure in pipeline logs-qualys_vmdr.asset_host_detection-5.5.0 failed with message: error message set and no data to process]
      qualys_vmdr/asset_host_detection fail-mismatch-xsd:
      [0] found error.message in event: [xsd and response mismatch:<?xml version="1.0" encoding="UTF-8"?>
      <!DOCTYPE HOST_LIST_VM_DETECTION_OUTPUT SYSTEM "https://qualysapi.qualys.com/api/2.0/fo/asset/host/vm/detection/dtd/output.dtd">
      <HOST_LIST_VM_DETECTION_OUTPUT>
        <TEST>
          <NAME>"XML Decode Test 1"</NAME>
          <DESCRIPTION>"THIS TEST SHOULD FAIL AS IT DOESN'T MATCH WITH XSD"</DESCRIPTION>
        </TEST>
      </HOST_LIST_VM_DETECTION_OUTPUT> Processor conditional with tag cel_failure in pipeline logs-qualys_vmdr.asset_host_detection-5.5.0 failed with message: error message set and no data to process]
      
      
      ╭─────────────┬──────────────────────┬───────────┬───────────────────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬───────────────╮
      │ PACKAGE     │ DATA STREAM          │ TEST TYPE │ TEST NAME         │ RESULT                                                                                                                          │  TIME ELAPSED │
      ├─────────────┼──────────────────────┼───────────┼───────────────────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┼───────────────┤
      │ qualys_vmdr │ asset_host_detection │ system    │ default           │ PASS                                                                                                                            │ 38.246982667s │
      │ qualys_vmdr │ asset_host_detection │ system    │ fail-invalid-xml  │ FAIL: test case failed: one or more errors found in documents stored in logs-qualys_vmdr.asset_host_detection-36902 data stream │ 38.958245875s │
      │ qualys_vmdr │ asset_host_detection │ system    │ fail-mismatch-xsd │ FAIL: test case failed: one or more errors found in documents stored in logs-qualys_vmdr.asset_host_detection-60407 data stream │ 41.050556708s │
      ╰─────────────┴──────────────────────┴───────────┴───────────────────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴───────────────╯
      --- Test results for package: qualys_vmdr - END   ---
      Done
      Error: failed to process results: one or more test cases failed
      ```

## Related issues


- Relates https://github.com/elastic/integrations/issues/11673

## Screenshots


**Captured errors in system tests:**

<img width="1356" alt="Screenshot 2024-12-05 at 4 38 11 PM" src="https://github.com/user-attachments/assets/d33e0cac-476d-4e10-ab03-37225512f834">
